### PR TITLE
Solaris Support

### PR DESCRIPTION
**Pull request by chris-rock**
- tested against Solaris 10, Solaris 11, SmartOS and OmniOS
- implements `file`, `os`, `package`, `service`, `group`, `user`, `post`, `etc_group` for Solaris

Test against Solaris 10:

```
$ KITCHEN_LOCAL_YAML=.kitchen.chef.yml b kitchen verify default-chef-solaris-1011
-----> Starting Kitchen (v1.5.0)
-----> Verifying <default-chef-solaris-1011>...
"You are currently running on OS: solaris, OS release: 10, OS family: solaris"
.............................................................................................

Finished in 2.66 seconds (files took 2.04 seconds to load)
96 examples, 0 failures

       Finished verifying <default-chef-solaris-1011> (0m4.57s).
-----> Kitchen is finished. (0m6.89s)
```

Test against Solaris 11:

```
$ inspec exec test/integration/test/integration/default  -t ssh://inspec@192.168.59.103 --password password --sudo --sudo-password=password
"You are currently running on OS: solaris, OS release: 11, OS family: solaris"
..............................................................................
...............

Finished in 2.34 seconds (files took 1.32 seconds to load)
96 examples, 0 failures

```

Test against OmniOS

```
$ inspec exec test/integration/test/integration/default -t ssh://vagrant@localhost:2201 --password vagrant --sudo-password=vagrant
"You are currently running on OS: omnios, OS release: 11, OS family: omnios"
.......................................................................................

Finished in 0.52465 seconds (files took 0.7729 seconds to load)
87 examples, 0 failures

```

Test against SmartOS

```
$ inspec exec test/integration/test/integration/default -t ssh://root@localhost:2202 --password vagrant
"You are currently running on OS: smartos, OS release: 11, OS family: smartos"
.......................................................................................

Finished in 1.64 seconds (files took 2.89 seconds to load)
87 examples, 0 failures

```

### CODE REVIEW

File: lib/resources/user.rb [hunk 1]
```diff
@@ -53,40 +53,42 @@ def initialize(user)
...
   end
 
   def exists?
-    !identiy.nil? && !identiy[:user].nil?
+    !identity.nil? && !identity[:user].nil?
   end
 
   def uid
-    identiy.nil? ? nil : identiy[:uid]
+    identity.nil? ? nil : identity[:uid]
```
Comment (srenatus) [inline lib/resources/user.rb]: What about `identity && identity[:uid]`?

File: lib/resources/user.rb [hunk 2]
```diff
@@ -53,40 +53,42 @@ def initialize(user)
...
+    elsif os.solaris?
+      @user_provider = SolarisUser.new(inspec)
     else
       return skip_resource 'The `user` resource is not supported on your OS yet.'
     end
   end
 
   def exists?
-    !identiy.nil? && !identiy[:user].nil?
+    !identity.nil? && !identity[:user].nil?
```
Comment (srenatus) [inline lib/resources/user.rb:75]: How about `identity && identity[:user]`? I think that's the same...

File: lib/resources/user.rb (hunk 1 above)
Comment (chris-rock) [inline lib/resources/user.rb]: +1

File: lib/resources/user.rb (hunk 2 above)
Comment (arlimus) [inline lib/resources/user.rb:75]: that last `identity[:user]` would not return a boolean; at least keep `.nil?` there imho

**Comment by arlimus** [comment]
Thank you for the huge inclusion for Solaris @chris-rock !!
:rocket:

File: lib/resources/user.rb (hunk 2 above)
Comment (srenatus) [inline lib/resources/user.rb]: That's fine, neither will `&&` ;)

```interactive
[2] pry(main)> foo = {}
=> {}
[3] pry(main)> foo[:bar] = 3
=> 3
[4] pry(main)> foo && foo[:bar]
=> 3
```

Comment (srenatus) [inline lib/resources/user.rb:75]: Ah, ok. You want this to be boolean, not just _truthy_.

**Pull request merged**